### ISSUE TITLE
Replace deprecated translation methods

### ISSUE THREAD

**Issue by juliusknorr** [nextcloud/text#6149]
Error: lib/Service/InitialStateProvider.php:28:3: DeprecatedInterface: Interface OCP\Translation\ITranslationManager is marked as deprecated (see https://psalm.dev/152)
Error: lib/Service/InitialStateProvider.php:62:4: DeprecatedClass: Class OCP\Translation\LanguageTuple is marked as deprecated (see https://psalm.dev/098)

https://github.com/nextcloud/documentation/pull/12063/files#diff-aaf8198084ab58336e1610260ae3ca96a1afee796cb21a305d03f2bf442948ea
https://github.com/nextcloud/server/pull/46775

@marcelklehr Do the old APIs still work or do we need to move this for 30?

**Comment by juliushaertl** [comment]
Do you think we need to keep both APIs in parallel then or just just migrating to the new one fine?

**Comment by marcelklehr** [comment]
I would just migrate to the new one directly.

**Comment by juliushaertl** [comment]
- [ ] Migrate the [FakeTranslationProvider](https://github.com/nextcloud/server/blob/afa48a4e0ed4515aec458db13a07e11a7e0f5981/apps/testing/lib/Provider/FakeTranslationProvider.php#L12) to the new task api as it is used by our cypress tests
- [ ] Migrate text initial state for translations to new task processing API

---
Notes from chat with Marcel:

- So, the Translate task type defines input and output language slots: https://github.com/nextcloud/server/blob/master/lib/public/TaskProcessing/TaskTypes/TextToTextTranslate.php#L69-L85
- these are defined as enums, the value for which is defined by the provider
- getAvailableTaskTypes contains the possible values in the key inputShapeEnumValues: https://github.com/nextcloud/server/blob/master/lib/public/TaskProcessing/IManager.php#L48-L52
- integration openai implements a translate provider for example: https://github.com/nextcloud/integration_openai/blob/main/lib/TaskProcessing/TranslateProvider.php
- translate2 app will implement this
- translate app not
- deepl/libretranslate need scheduling

**Comment by julien-nc** [comment]
Hey there,

The testing app now has a task processing translation provider: `OCA\Testing\TaskProcessing\FakeTranslateProvider`

More details on how to run translation tasks and get the available languages in the `translate` section of https://github.com/nextcloud/assistant/issues/114
